docs/basic-demos

KacperMadej opened the pull request:
Added Basics demo category.

### To do
 - [x] General alignment with designs. Rounded plot border etc.
 - [x] Tooltips (make some of the demos use HTML and SVG icons like the designs)
 - [x] Make `demo.ts` where possible
 - [x] General styling for `.highcharts-description`
 - [x] Scatter demo: friendlier data handling. Consider CSV loaded into a DataTable. See if we can find newer data (like in the Violin plot).
 - [x] Heatmap demo: What happened to the color axis marker? What's up with white data labels on bright tiles?
 - [x] Candlestick demo: Missing volume pane
 - [x] Gantt: push more for the proposed design sketch. Visible tree grid, top grid axis, assignee mugshot. Use DataTable?
 - [x] Basic sparklines: Make live

#### To consider
* Tooltip defaults to work better with the designs. [Draft PR created](https://vrevs.highsoft.com/pr/24592/review), but there are some fundamental problems like split tooltips.
* Navigator outline with rounded corners and extending all the way around.
* More color axis marker options to allow the design proposal.

Postponed:
- new designs for Dashboards

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1213608062726189
  - https://app.asana.com/0/0/1209013878475682

KacperMadej committed:
Added Basics demo category.

KacperMadej committed:
Added minimal changes for adaptive dark mode.

oysteinmoseng reviewed (approved):
Very nice, leaving it to Torstein to take it further for tweaks to the demo content 👍

TorsteinHonsi commented on samples/highcharts/demo/bar-chart/demo.js:
```diff
@@ -41,20 +40,20 @@ Highcharts.chart('container', {
-        borderWidth: 1,
-        backgroundColor: 'var(--highcharts-background-color, #ffffff)',
-        shadow: true
+        floating: true
```
We need to keep the background, otherwise the grid lines cut right through the legend. Can be seen in the visual diff.
```suggestion
        floating: true,
        backgroundColor: 'var(--highcharts-background-color, #ffffff)'
```

TorsteinHonsi commented on samples/highcharts/demo/donut/demo.css:
```diff
@@ -0,0 +1,60 @@
+
+.highcharts-data-table table {
+    font-family: Verdana, sans-serif;
+    border-collapse: collapse;
+    border: 1px solid #ebebeb;
+    margin: 10px auto;
+    text-align: center;
+    width: 100%;
+    max-width: 500px;
+}
+
+.highcharts-data-table caption {
+    padding: 1em 0;
+    font-size: 1.2em;
+    color: #555;
+}
+
+.highcharts-data-table th {
+    font-weight: 600;
+    padding: 0.5em;
+}
+
+.highcharts-data-table td,
+.highcharts-data-table th,
+.highcharts-data-table caption {
+    padding: 0.5em;
+}
+
+.highcharts-data-table thead tr,
+.highcharts-data-table tr:nth-child(even) {
+    background: #f8f8f8;
+}
+
+.highcharts-data-table tr:hover {
+    background: #f1f7ff;
+}
+
+.highcharts-description {
+    margin: 0.3rem 10px;
+}
```
This CSS is redundant, the demo doesn't use export-data.js

TorsteinHonsi commented on samples/highcharts/demo/donut/demo.js:
```diff
@@ -0,0 +1,39 @@
+Highcharts.chart('container', {
+    chart: {
+        type: 'pie',
+        backgroundColor: '#222'
```
@oysteinmoseng Regarding explicit dark mode demos - should we continue having that now that we have the adaptive theme and a light/dark switch in the UI? The pro is that we get to display theming at a glance. The con is that this demo is locked in dark mode even if users opt for light mode.

TorsteinHonsi replied:
If we choose to go for explicit dark mode, the background color should be `#141414` to match the adaptive theme.

oysteinmoseng replied:
I think it's important that not all of these basic demos use the same theme - we need to show a wide array of styling. We should however also show off best practice for theming. With all the recent changes I am not actually sure what the best practice is (presumably something that works with light & dark mode), so maybe you can chime in on that part...

TorsteinHonsi commented on samples/highcharts/demo/gauge/demo.css:
```diff
@@ -0,0 +1,60 @@
+    margin: 1em auto;
+}
+
+.highcharts-data-table table {
+    font-family: Verdana, sans-serif;
+    border-collapse: collapse;
+    border: 1px solid #ebebeb;
+    margin: 10px auto;
+    text-align: center;
+    width: 100%;
+    max-width: 500px;
+}
+
+.highcharts-data-table caption {
+    padding: 1em 0;
+    font-size: 1.2em;
+    color: #555;
+}
+
+.highcharts-data-table th {
+    font-weight: 600;
+    padding: 0.5em;
+}
+
+.highcharts-data-table td,
+.highcharts-data-table th,
+.highcharts-data-table caption {
+    padding: 0.5em;
+}
+
+.highcharts-data-table thead tr,
+.highcharts-data-table tr:nth-child(even) {
+    background: #f8f8f8;
+}
+
+.highcharts-data-table tr:hover {
+    background: #f1f7ff;
+}
```
Redundant data table CSS

TorsteinHonsi commented on samples/highcharts/demo/gauge/demo.js:
```diff
@@ -0,0 +1,78 @@
+    chart: {
+        type: 'gauge',
+        height: 300,
+        spacingBottom: -150
```
Use `pane.center` and `pane.size` instead, it's responsive.

TorsteinHonsi commented on samples/highcharts/demo/pie-chart/demo.js:
```diff
@@ -1,77 +1,44 @@
+            format: '{point.percentage:.0f}%',
+            distance: -30, // Placing the label inside
+            style: {
+                textOutline: 'none'
```
There's room for larger fonts here, so we don't need to hold back
```suggestion
                fontSize: '0.9em',
                textOutline: 'none'
```

TorsteinHonsi reviewed (changes requested):
Thanks, that's a nice overview! 

Apart from the inline comments, I'm wondering about the Stock, Maps and Gantt demos. Are the ones marked "Basics" supposed to go into the core Basics group? If so, they don't. Probably because they are under a separate product. Also, the Dashboards product has a Basic category (singular). So we should sort out these things. In the release procedure, point 3c, see how to test these things with the demo pages.

oysteinmoseng commented:
> Are the ones marked "Basics" supposed to go into the core Basics group? If so, they don't. Probably because they are under a separate product. Also, the Dashboards product has a Basic category (singular).

Yes, I would say the main purpose of the new Basics group is to have a non-product-specific, above the fold, overview. Should definitely include one or more demos from each product. Thinking about it, it's not ideal if this category is within Highcharts Core. It would be nice to have a product group that is "Overview" or something.

I don't mind that the Dash product has a specific "Basic" category (meaning "Dashboard Basics" vs "Overall Basics"), does it cause any issues?

KacperMadej committed:
Review changes. Donut candidate. New demos section.

KacperMadej committed:
Fixed demo manager tags.

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

KacperMadej committed:
Added Grid sparklines to Basics.

KacperMadej committed:
Added note about # in the docs redirect.

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

KamilKubik commented on samples/demo-config.js:
```diff
@@ -7,6 +7,11 @@
  * title
  */
 module.exports = {
+    Overview: {
```
```suggestion
    'Highcharts Overview': {
```

KamilKubik reviewed (changes requested):
Looks good, I tested the demo pages build, and that's the only change I'd make to match our current architecture (much easier from the demo pages perspective), thanks!

KacperMadej committed:
Update samples/demo-config.js

Co-authored-by: Kamil Kubik <50461863+KamilKubik@users.noreply.github.com>

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

KamilKubik reviewed (approved):
Looks good!

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

KacperMadej committed:
Updated design: bar.

KacperMadej committed:
Updated design: line.

KacperMadej committed:
Updated design: column.

KacperMadej committed:
Updated design: pie & donut.

KacperMadej committed:
Linted.

KacperMadej committed:
Update design: gauge.

KacperMadej committed:
Update design: bubble.

KacperMadej committed:
Update design: scatter.

KacperMadej committed:
Update design: heatmap.

KacperMadej committed:
Update design: candlestick.

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

KacperMadej committed:
Linted.

KacperMadej committed:
Update design: maps.

KacperMadej committed:
Gantt demo changed and some fixes.

KacperMadej committed:
Linted.

KacperMadej committed:
Linted.

KacperMadej committed:
Linted.

KacperMadej committed:
Linted.

stitot committed:
Grid Pro basic demo

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

stitot committed:
Dashboards demo  initial commit

stitot committed:
Final basic Dashboards demo

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

KacperMadej committed:
Merge branch 'master' into docs/basic-demos

TorsteinHonsi commented:
@KacperMadej @oysteinmoseng Sorry for the delay in reviewing this, but I have been waiting for the new design-related capabilities for v13 (rounded plot area, tick marks as dots...). I added this PR to the v13 project as I think it will be a nice fit for the v13 push.

TorsteinHonsi commented:
@KacperMadej 

> I have been waiting for the new design-related capabilities for v13

Meanwhile, can we work on everything else but the final design? Setting up boilerplates for all the demos (as close as we get to the design target) and making them appear in the right place in the Basic category?

TorsteinHonsi committed:
Basic demos: Partly bar-chart
New design was too text-heavy at the bottom, while top right had large open space

TorsteinHonsi committed:
Basic demos: Pie chart to match updated deign

TorsteinHonsi committed:
Basic demos: Heatmap colorAxis styling

TorsteinHonsi committed:
Merge branch 'trettan' into docs/basic-demos

TorsteinHonsi committed:
Basic demos: alignment with design sketches

TorsteinHonsi committed:
Basic demos: TypeScript where applicable

TorsteinHonsi committed:
Demos: removed unused readme.md files
We're now using descriptions in demo.html instead

TorsteinHonsi committed:
Samples: removed unused readme.md files
We're now using descriptions in demo.html instead

TorsteinHonsi committed:
Basic demos: styled description paragraph

TorsteinHonsi committed:
Basic demos: tooltips

TorsteinHonsi committed:
Samples: use DataTable for candlestick-and-volume

TorsteinHonsi committed:
Basic demos: cleaner scatter

TorsteinHonsi committed:
--amend

TorsteinHonsi committed:
Pre-release regression: color axis marker clipped when plotBorderRadius

TorsteinHonsi committed:
Basic demos: Added options and demo for color axis marker

TorsteinHonsi committed:
Basic demos: grid axis on top

TorsteinHonsi committed:
Samples: nicer candlestick tooltip

TorsteinHonsi committed:
Samples: gantt demo

ddragula committed:
Added live update for the basic sparklines demo.

ddragula committed:
Fixed interval stopping on responsive change.

TorsteinHonsi committed:
Samples: use common description styling

TorsteinHonsi committed:
Samples: naming in the basic demos

TorsteinHonsi committed:
Samples: naming

TorsteinHonsi committed:
Merge pull request #24705 from highcharts/docs/basic-sparklines-live-update

docs/basic-sparklines-live-update

TorsteinHonsi committed:
Samples: sparklines adjustments
